hoomd/md/manifold.py

```diff
@@ -0,0 +1,407 @@
+# Copyright (c) 2009-2019 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+# Maintainer: joaander / All Developers are free to add commands for new features
+
+import hoomd;
+from hoomd.md import _md
+from hoomd import _hoomd
+from hoomd.operation import _HOOMDBaseObject
+from hoomd.data.parameterdicts import ParameterDict, TypeParameterDict
+from hoomd.data.typeconverter import OnlyIf, to_type_converter
+from collections.abc import Sequence
+
+## \internal
+# \brief Base class for manifold
+#
+# A manifold in hoomd reflects a Manifold in c++. It is respnsible to define the manifold 
+# used for RATTLE intergrators and the active force constraints.
+class Manifold(_HOOMDBaseObject):
```

Comment:
Should the name be `ConstraintManifold` since these manifolds only serve that purpose?

Reply:
These classes are fairly general descriptions of surfaces that are manifolds. It is conceivable that we use them for other purposes than constraints in the future.